### ISSUE TITLE
Feature Pitch: Make gold blocks visible on minimap 

### ISSUE THREAD

**Issue by bunniewormy** [transhumandesign/kag-base#126]
it's a bit stupid that you have to dig, dig and keep digging on big, weird maps if you need some of that beautiful siege and tunnel material called GOLD. On some maps it's in weird places you'd never expect it to be in, sometimes there's not even any gold left on the map.

And you have no way of knowing that. Similar with stone.

The worse is, that you could actually find out - and not many people know about it; you'd have to open console and type /savemap mapname. You could actually go to kag folder / base / maps and see in the mapname's png where the goddamn gold and stone is. It gives people who know about it an unfair advantage; although there probably aren't many.

It would make sense to fix it then; make gold show on minimap. That way players would know where to dig to get a material that is actually pretty important [siege & tunnels].

Now, you could say that it's too big of a change. That you're supposed to keep digging and digging until you find the gold somewhere.

That argument is stupid; you currently have to memorise where shit is on every map. Or you could just randomly decide where to dig if you want to find gold: keep digging in direction A, or maybe B or C. Finding gold shouldn't depend on luck.

To these saying it's a problem with the maps - is it? Technically, you could put caves in a map and potentially put gold into there, but then, are you going to make every cave on every map have gold? if not, then you still could dig into a cave you saw on the minimap and find nothing. Because underground caves aren't supposed to just have gold in them.

TL;DR:
- Gold is in weird places on maps. It's not a problem with the maps. Gold should be underground.
- You shouldn't be forced to dig in random directions and find gold by luck. It's an important material needed for siege and tunnels.
- You should have a way of knowing if there is even any gold left on the map.
- This is going to be just a simple QoL convenience change. You can already /savemap in the console and check the map's PNG file, but that's boring and unfair.

My suggestion is pretty simple: make gold blocks  visible on the minimap.

**Comment by AcrylicOracle** [comment]
it's not unfair to look at the map >:( but i do agree to some extent that there should be an easier way to find gold with maps that have lots of underground

**Comment by Orsovi** [comment]
Well if you do so with gold, you'll have to do it with with others ressources like stone. We'll fast have a minimap with too much information.
For me not easily-accessible gold is not a real problem, it give a direction to the game depending on the map and that's interesting. If you don't have siegeshop or tunnel that's not a problem ! And if someone find the gold, it can push the other team to look for it !

So maybe : 
- To mapmaker : dont put gold in stupid place (a gold mine at the middle is **always** a good idea)
- Why not add a note if there isn't no more gold on the map but not localize it on the minimap.

**Comment by Suprejso** [comment]
Sorry but I can't agree with it. Gold, as the most rare resource on the map should be hard to find, making it visible on the minimap will make it too easy to get, I can already see people just digging in specific places and then making a ton of vehicles like catapults or ballistas. Not a good idea in my opinion.

**Comment by Coroz** [comment]
Should be visible and most maps need the gold reduced.. so you have to choose between tunnel or siege at the start of the game. I found gold on the wierdest locations when i was drilling beneath my base for a bit of fun.

**Comment by bunniewormy** [comment]
"Gold should be hard to find"
Except that gold isn't even hard to find - it's not like you gotta be skilled or anything to find it; it's just frustrating, boring and depending on luck too much. And if you memorised the gold placement on a map then it's not 'hard to find' anymore and you can dig in specific places... Plus as i said, /savemap. I feel like people are missing the point on here

**Comment by bunniewormy** [comment]
Also, you don't really need to put stone on minimap. I mean, I wouldn't mind it, but stone isn't such an important resource as gold, especially because you can get it from resupplies.

**Comment by Suprejso** [comment]
First you say that gold isn't hard to find but just frustrating, boring and depending on luck too much and then "if you memorised the gold placement on a map then it's not 'hard to find' anymore(...)"
It's more like the gold placement should be changed, no need to make it visible on minimap, at least that's how I see it after I read your replies.
Another point of view: imagine clanwars with this system. One team will just build a tunnel at base, send one builder who'll rat and get all the gold as its easy to get it because its visible on the minimap, right? Then he just builds a tunnel after getting the flag and that's it.

**Comment by MrPotatobird** [comment]
Think of it this way: You want some gold. You don't see any. You don't know where it is or might be because you have not memorized all the maps. You don't know if there even is any. You start digging around in random patches of dirt. You do not find the gold. Maybe you were close, but because you were just digging around randomly you didn't find it.

Is that fun?

It's not "hard" to get, it's annoying and random if you don't know where it is. That's not the kind of fun challenge that we like kag for.

Don't worry about clanwars, because if people are tryharding they'll just open the map.png or do /savemap or something so they can see exactly where the gold is on any map, and then get the gold and build their tunnels or whatever (that already happens and tunnels are a fun part of competitive.) But generally people shouldn't be expected to go through those annoyances.

**Comment by bunniewormy** [comment]
"Another point of view: imagine clanwars with this system. One team will just build a tunnel at base, send one builder who'll rat and get all the gold as its easy to get it because its visible on the minimap, right? Then he just builds a tunnel after getting the flag and that's it."

I've looked at the map png, tried this, and it didn't work. Getting the gold should be at best a time effort that is equal for both teams [and not have 1 team luckier by digging in a different direction]. Plus having gold doesn't even make you instantly win the game. And usually, competitive maps have gold in visible places as well.

**Comment by Coroz** [comment]
There shouldn't be a effort with finding the first 50 to 100 gold tho..(if there is gold on the map). otherwise you make matches unnecessary unbalanced. Where old fags know the locations and newbies don't.

**Comment by Biurza** [comment]
I like the idea.
This makes up for a good balance since if, excuse me, but i am not as much of a tryhard as someone that has another tab with map png file open and/or has gold placement memorized and rolled in just to have quick fun, i shouldn't be struggling with his siege/tunnels/whatever while digging whole map in search of these few blocks which are just placed in a weird place.

**Comment by AcrylicOracle** [comment]
this isn't a problem. most of the gold you need can be found in relatively shallow/open areas in the maps that you see most often in pubs. as said far above, some tryhards will open up the map png or do savemap, but the essentials of finding gold is that by the time that you really need the gold, most of it will already be surfaced.
      usually people only find a need for gold when a stalemate is reached, and by that time a lot of the map should be blown away/ mined out, to the point where digging for gold isn't even an option. most or all of the gold is either surfaced or found. and i've found the maps that people are looking for gold in the most are the maps where all of the gold is already surfaced, which is just bad map design. there's no point in complaining about it, because like i said, gold is pretty much a stalemate winner. it's a commodity, not a necessity. so why make the developers change that?

**Comment by bunniewormy** [comment]
said the guy who looks into .pngs at clanwars
and didnt you previously say "but i do agree to some extent that there should be an easier way to find gold with maps that have lots of underground"?

**Comment by MrPotatobird** [comment]
It's a feature pitch, not an issue; it would just be a nice little quality of life addition, so there's no use talking about how it's "not that big of a problem." I mean, what are you worried about, making the game better for nothing?

Also, the nice thing about this here repo is that we're not "making" the developer change anything, one of us could try to figure it out. If the devs wanna put this on their list then cool, if there's other higher priorities then that's fine.

**Comment by 1bardesign** [comment]
Addendum: 
```cpp
//apply flood layers 
const SColor water_color(0xff2cafde);
const SColor fire_color(0xffd5543f);

if (map->isInWater( off ) )
{
	col = col.getInterpolated(water_color,0.5f);
}
else if (map->isInFire( off ) )
{
	col = col.getInterpolated(fire_color,0.5f);
}
```
(this should be applied after determining the correct colour as above)

**Comment by 1bardesign** [comment]
Done by #167 thanks @Pirate-Rob and @GoldenGuy
